### DISCUSSION TITLE
Player installation failed | Ed Easy Installer

### DISCUSSION THREAD

**Discussion by Marc70008** [Sunbird-Ed/Community#542] [Installation] [answered]
Hi, I am following the Ed Easy Installer script to provision Ed platform
While running the Ed easy installer bash script, Player(ED BB) installation failed due to `template: player/templates/configmap.yaml:99:165: executing "player/templates/configmap.yaml" at <.Values.external_secrets.keycloak_realm_public_key.name>: nil pointer evaluating interface {}.keycloak_realm_public_key`

I have observed that the external secret: keycloak_realm_public_key, are not found while installing player helm.I am not sure where this is referenced from.
Because of the player installation failure, nginx-public-ingress is in crashloopbackoff with error `host not found in upstream "player:3000" in /etc/nginx/nginx.conf`

How do we fix this?

**Comment by Krishnaj20** [comment]
@gandham-santhosh - Can you check and respond to this thread please?

**Comment by tsprasath** [comment]
@Marc70008  After post install nginx should be up and running. Please re-run the post-install and check it.

`./install-ed.sh <<kubeconfig>> -i postscript`

**Reply by Marc70008** [reply]
@tsprasath , i reran the post script, all the tokens are fetched except for api-admin
Noticed error in the logs of the pod created by onboardconsumer
error:
`Traceback (most recent call last):
  File "kong_consumers.py", line 159, in <module>
    save_consumers(args.kong_admin_api_url, input_consumers)
  File "kong_consumers.py", line 54, in save_consumers
    _save_groups_for_consumer(kong_admin_api_url, consumer)
  File "kong_consumers.py", line 145, in _save_groups_for_consumer
    json_request("POST", consumer_acls_url, {'group': input_group})
  File "/app/config/..2023_07_05_06_04_56.2482128959/common.py", line 36, in json_request
    response = retrying_urlopen(request)
  File "<decorator-gen-2>", line 2, in retrying_urlopen
  File "/usr/lib/python2.7/site-packages/retry/api.py", line 74, in retry_decorator
    logger)
  File "/usr/lib/python2.7/site-packages/retry/api.py", line 33, in __retry_internal
    return f()
  File "/app/config/..2023_07_05_06_04_56.2482128959/common.py", line 41, in retrying_urlopen
    return urllib2.urlopen(*args, **kwargs)
  File "/usr/lib64/python2.7/urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "/usr/lib64/python2.7/urllib2.py", line 437, in open
    response = meth(req, response)
  File "/usr/lib64/python2.7/urllib2.py", line 550, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib64/python2.7/urllib2.py", line 475, in error
    return self._call_chain(*args)
  File "/usr/lib64/python2.7/urllib2.py", line 409, in _call_chain
    result = func(*args)
  File "/usr/lib64/python2.7/urllib2.py", line 558, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
urllib2.HTTPError: HTTP Error 400: Bad Request`

Because of the above, pods created during post install are failing.

How do we fix this?

**Reply by tsprasath** [reply]
@Marc70008  Please execute below command in bash .. it will extract the token for api-admin

`LOGS=$(kubectl logs -l job-name=onboardconsumer -n dev --tail=10000 | grep -E "JWT token for api-admin is")`
`TOKEN=$(echo $LOGS | grep -oP "(?<=: ).*")`
`echo "JWT token for api-admin:"`
`echo "$TOKEN"`
`echo "core_vault_sunbird_api_auth_token: \"$TOKEN\"" >> global-values.yaml`
`echo "sunbird_api_auth_token: \"$TOKEN\"" >> global-values.yaml`

**Reply by Marc70008** [reply]
@tsprasath , no string "JWT token for api-admin is" in the job/pod logs.
but the other JWT token strings for  adminutil_learner_api_key,  portal_anonymous_register etc. are found.

**Reply by Marc70008** [reply]
[onboardconsumer-error-log.txt](https://github.com/Sunbird-Ed/Community/files/11957684/onboardconsumer-error-log.txt)
FYI, @tsprasath

**Reply by tsprasath** [reply]
@Marc70008 it is already present 
```
Adding group badgeCreate for consumer api-admin
{"group":"ACL group already exist for this consumer"}
```
Please check how many time onboardconsumer jobs has ran.

**Reply by Marc70008** [reply]
@tsprasath ,this has been run only once, and the status is ERROR, 
the value of "core_vault_sunbird_api_auth_token" is also not updated in global-valueys.yaml

**Reply by tsprasath** [reply]
@Marc70008 Please try the below command 

```
helm uninstall onboardconsumer -n dev
sleep 30
helm upgrade --install onboardconsumer https://sunbirdartifact.blob.core.windows.net/easyinstaller/onboardconsumer-0.1.0.tgz -n dev
```

Later try this command

```
LOGS=$(kubectl logs -l job-name=onboardconsumer -n dev --tail=10000 | grep -E "JWT token for api-admin is")
TOKEN=$(echo $LOGS | grep -oP "(?<=: ).*")
echo "JWT token for api-admin:"
echo "$TOKEN"
echo "core_vault_sunbird_api_auth_token: \"$TOKEN\"" >> global-values.yaml
echo "sunbird_api_auth_token: \"$TOKEN\"" >> global-values.yaml
```

**Reply by Marc70008** [reply]
@tsprasath , _kubectl logs -l job-name=onboardconsumer -n dev --tail=10000 | grep -E "JWT token for api-admin is"_ returned empty sting
i can still see the 400 bad request error in log output
There is no stirng "JWT token for api-admin is" in the log output
Shared the log outpur for reference
[onboarconsumer_job_output.txt](https://github.com/Sunbird-Ed/Community/files/12013321/onboarconsumer_job_output.txt)

**Reply by tsprasath** [reply, marked as answer]
@Marc70008 

We have seen this occur occasionally. Please login to postgres kong database and run truncate the acls table.

```
truncate acls;
```
```
helm uninstall onboardconsumer -n dev
sleep 30
helm upgrade --install onboardconsumer https://sunbirdartifact.blob.core.windows.net/easyinstaller/onboardconsumer-0.1.0.tgz -n dev
```
```
LOGS=$(kubectl logs -l job-name=onboardconsumer -n dev --tail=10000 | grep -E "JWT token for api-admin is")
TOKEN=$(echo $LOGS | grep -oP "(?<=: ).*")
echo "JWT token for api-admin:"
echo "$TOKEN"
echo "core_vault_sunbird_api_auth_token: \"$TOKEN\"" >> global-values.yaml
echo "sunbird_api_auth_token: \"$TOKEN\"" >> global-values.yaml
```

**Reply by Marc70008** [reply]
@tsprasath , when i logged into the postgresql pod, couldnot find a database name kong while listing databases.
![image](https://github.com/Sunbird-Ed/Community/assets/72565751/2fc971c3-41fb-4e2d-96b6-902cdb2c854f)

**Reply by tsprasath** [reply]
@Marc70008  `api_manager_dev` is the database

**Reply by Marc70008** [reply]
@tsprasath , able to fetch the api-admin token, will rerun postinstall script and update

**Reply by tsprasath** [reply]
@Marc70008 if the issue resolved please close the ticket

**Comment by Marc70008** [comment]
run _"truncate acls"_ in api_manager_dev Database in postgres pod. 
This solved the issue
